### ISSUE TITLE
[dev.icinga.com #7934] Include a text cue for required form control labels

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icingaweb2#1221]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/7934

**Created by elippmann on 2014-12-03 10:05:46 +00:00**

Assignee: _jmeyer_
Status: _Resolved_  (closed on _2015-02-26 15:05:43 +00:00_)
Target Version: _2.0.0-beta3_
Last Update: _2015-02-26 15:05:43 +00:00 (in Redmine)_

---

http://www.w3.org/TR/WCAG20-TECHS/general.html#G205 wrote:
> The objective of this technique is to combine color and text or character cues to convey information. Most users can quickly scan the content to locate information conveyed by using color differences. Users who cannot see color can look or listen for text cues; people using Braille displays or other tactile interfaces can detect text cues by touch.
> 
> The text cue must be included as part of the programmatically determinable name for the control.

When setting a form element required in our code, its label must have the CSS class <code>required</code> added automatically. For screen readers the cue "(required)" must be included within the label. By default its the asterisk "\*".

**Changesets**

_2015-02-06 16:50:34 +00:00_ by _aklimov_ 58d48d9fa048c673b63903291afdc4b10d70825c

    Accessibility: Text cue for required form control labels: Add prototype
    
    refs #7934

_2015-02-12 08:09:18 +00:00_ by _jmeyer_ 60ec1157797623acfeb8e1b67d3d1cc0906e867b

    WCAG/1.4.1: Fix example for text cues on a form's required elements
    
    The input element must be wrapped by <label> either or the <label> tag
    needs a "for=id" attribute with the id of the respective input.
    
    refs #7934

_2015-02-12 08:11:02 +00:00_ by _jmeyer_ c31cab39cd30f196186beb6aa3eb7435e80ac930

    WCAG/1.4.1: Fix example for text cues on a form's required elements
    
    The input element must be wrapped by <label> either or the <label> tag
    needs a "for=id" attribute with the id of the respective input.
    
    refs #7934

_2015-02-26 15:05:13 +00:00_ by _jmeyer_ 0111fdf78ea9783d5ce2506001473c2ec7db6ad5

    Add an asterisk suffix to a form element's label if its required
    
    The screen reader part "(required)" is not being considered with this
    commit as most screen readers do not need this due to the ARIA and
    HTML5 property.
    
    resolves #7934

_2015-02-27 08:08:05 +00:00_ by _jmeyer_ c00e336ea8ce308a2e2077ee4ae3e750c7c6ec9f

    Make the cue that is being appended to required form elements configurable
    
    refs #7934

_2015-02-27 08:09:25 +00:00_ by _jmeyer_ ea4de8254586a222a1f4282f826219bc8ffd100f

    Login: Disable the required cue
    
    It should be obvious that it is necessary to input a name and a password
    to login in case you're looking at a "login" form...
    
    refs #7934

_2015-03-02 12:48:35 +00:00_ by _jmeyer_ 1eacaa4c48daad665b1a1e3b44c19cb7d865d2df

    Improve the display of the text cue on required form elements
    
    Shows just a message at the top of the form if all elements are required
    and a explaining message what the cue is standing for otherwise.
    
    refs #7934

_2015-03-02 17:40:46 +00:00_ by _jmeyer_ d714c4323c41ea8d799b8d15642e9e58d04750d1

    Installation: Remove required cue from the token input
    
    refs #7934

_2015-03-02 17:40:46 +00:00_ by _jmeyer_ ea7b77eb72264503f822cb41938e8410d2d13f08

    Remove the required cue when disabling notifications globally
    
    refs #7934

_2015-03-02 17:40:46 +00:00_ by _jmeyer_ d3f9bd633cd5666f1272e136291c9583221acdbd

    Installation: Hide the required cue when choosing the preference backend
    
    refs #7934
